tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -433,11 +433,13 @@ spec:
 
 			stdOut := helper.CmdShouldPass("odo", "link", "EtcdCluster/example")
 			Expect(stdOut).To(ContainSubstring("Successfully created link between component"))
+			helper.CmdShouldPass("odo", "push")
```

Comment:
From the logs https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-odo-master-v4.2-integration-e2e-periodic/1298409952288706560#1:build-log.txt%3A1119 I didn't see it failing over here. Things are breaking with the lines https://github.com/openshift/odo/pull/3837/files#diff-49f09fb5341717455c0421b3c592e685R449 . So I am not sure this will fix the exact issue.

Reply:
You're right. But adding `odo push` at the place where you're suggesting it would be just a hack to make sure that we introduce some timeout between the two calls. Adding it where I have proposed in the PR right now makes sure that we're following the flow that a user would when using `odo link`/`odo unlink` commands and it still gives us the timeout we need to avoid the race condition from arising.

Reply:
TBH I don't understand why we are bandaging it with `odo push` when its not necessary. I would say lets not complicate the test scenario, consider a simple example:

```
> odo link EtcdCluster/example
			
> odo unlink EtcdCluster/example

> odo link EtcdCluster/example  //This is failing

> odo push
```
Maybe your current changes will hide the issue but it won't fix the real issue. In future I am pretty sure somebody will hit the issue with the same scenario. Also frankly speaking as a end user I won't rely much on `.odo/env/env.yaml` for services state failure.